test/e2e/README.md

```diff
@@ -120,6 +120,8 @@ enabled explicitly.
 * To run the Prometheus tests within the e2e suite, use:
 `go test -v github.com/vmware-tanzu/antrea/test/e2e --prometheus`
 
+[Target metadata API](https://prometheus.io/docs/prometheus/latest/querying/api/#querying-target-metadata) is used to validate metrics in the e2e test.
+The API has been available since Prometheus v2.4.0 but it's still experimental in v2.19.3.
```

Comment:
I'm not sure this information is needed here and it may be a bit confusing. No action is required from users here, since `./infra/vagrant/push_antrea.sh --prometheus` will take care of deploying the correct version of Prometheus.

Reply:
That makes sense that this is not necessary information for users. Let me remove that change.